### PR TITLE
task-48748: disable kudos for disabled users

### PR DESCRIPTION
**Pull request by hthabti**
prior to this change it was possible to try to send kudos to disabled users
fix:
1) update on buttonDisabled to check if receiver's identity is enabled so if its value is false the action button on comments and activities would be disabled
2) display/not display kudos icon for users popover

### CODE REVIEW

File: kudos-webapps/src/main/webapp/vue-app/kudos/components/KudosButton.vue [hunk 1]
```diff
@@ -118,11 +118,19 @@ export default {
...
     buttonDisabled() {
       if (this.comment) {
-        const commentOwnerId = this.comment.identity && this.comment.identity.id;
-        return commentOwnerId === eXo.env.portal.userIdentityId;
+        const commentOwnerId = this.comment.identity && this.comment.identity.id; 
+        if (commentOwnerId === eXo.env.portal.userIdentityId) {
+          return 'same'; 
+        } else if (!this.comment.identity.profile.dataEntity.enabled){
+          return 'disabled';        
+        }
```
Comment (ahamdi) [inline kudos-webapps/src/main/webapp/vue-app/kudos/components/KudosButton.vue]: Use a boolean computed property and use it for the check , instead of checking on a String value 'disabled'

File: kudos-webapps/src/main/webapp/vue-app/kudos/components/PopoverKudosButton.vue [hunk 2]
```diff
@@ -29,7 +31,16 @@ export default {
...
+    identityEnabled: {
+      type: String,
+      default: '',
+    },      
   },
+  computed: {
+    enabled() {      
+      return this.identityEnabled;
+    }
+  },
```
Comment (ahamdi) [inline kudos-webapps/src/main/webapp/vue-app/kudos/components/PopoverKudosButton.vue]: No need to add the computed property here. use directly this.identityEnabled

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: kudos-webapps/src/main/webapp/vue-app/kudos/components/PopoverKudosButton.vue [hunk 3]
```diff
@@ -4,6 +4,8 @@
       <div
         v-bind="attrs"
         v-on="on">
+         <div v-if="!identityDeleted">  
+         <div v-if="identityEnabled">
```
Comment (ahamdi) [inline kudos-webapps/src/main/webapp/vue-app/kudos/components/PopoverKudosButton.vue]:
```suggestion
         <div v-if="identityEnabled && !identityDeleted">  
```

File: kudos-webapps/src/main/webapp/vue-app/kudos/components/PopoverKudosButton.vue [hunk 4]
```diff
@@ -12,6 +14,8 @@
           <v-icon size="18">fas fa-award</v-icon>
         </v-btn>
       </div>
+      </div>
+      </div>
```
Comment (ahamdi) [inline kudos-webapps/src/main/webapp/vue-app/kudos/components/PopoverKudosButton.vue]:
```suggestion
      </div>
```

File: kudos-webapps/src/main/webapp/vue-app/kudos/components/KudosButton.vue [hunk 5]
```diff
@@ -116,13 +116,30 @@ export default {
...
     buttonDisabled() {
       if (this.comment) {
-        const commentOwnerId = this.comment.identity && this.comment.identity.id;
-        return commentOwnerId === eXo.env.portal.userIdentityId;
+        const commentOwnerId = this.comment.identity && this.comment.identity.id; 
+        if (commentOwnerId === this.userIdentityId) {
+          return 'same'; 
+        } else if (this.inactiveCommentOwner){
+          return 'inactive';        
+        }
```
Comment (ahamdi) [inline kudos-webapps/src/main/webapp/vue-app/kudos/components/KudosButton.vue]:
```suggestion
        const commentOwnerId = this.comment.identity && this.comment.identity.id; 
        return commentOwnerId === this.userIdentityId || inactiveCommentOwner();
```

File: kudos-webapps/src/main/webapp/vue-app/kudos/components/KudosButton.vue [hunk 6]
```diff
@@ -116,13 +116,30 @@ export default {
...
       } else if (this.activity) {
-        const activityOwnerId = this.activity.identity && this.activity.identity.id;
-        return activityOwnerId === eXo.env.portal.userIdentityId;
+        const activityOwnerId = this.activity.identity && this.activity.identity.id; 
+        if (activityOwnerId === this.userIdentityId){
+          return 'same';
+        } else if (this.inactiveActivityOwner){
+          return 'inactive';        
+        }
```
Comment (ahamdi) [inline kudos-webapps/src/main/webapp/vue-app/kudos/components/KudosButton.vue]:
```suggestion
        const activityOwnerId = this.activity.identity && this.activity.identity.id; 
        return activityOwnerId === this.userIdentityId || inactiveActivityOwner();
```

File: kudos-webapps/src/main/webapp/vue-app/kudos/components/KudosButton.vue [hunk 7]
```diff
@@ -10,7 +10,7 @@
           v-on="on">
           <v-btn
             :id="`KudosActivity${entityId}`"
-            :disabled="buttonDisabled"
+            :disabled="buttonDisabled === 'same' || buttonDisabled === 'inactive'"
```
Comment (ahamdi) [inline kudos-webapps/src/main/webapp/vue-app/kudos/components/KudosButton.vue]:
```suggestion
            :disabled="buttonDisabled"
```

File: kudos-webapps/src/main/webapp/vue-app/kudos/components/KudosButton.vue (hunk 1 above)
Comment (hthabti) [reply kudos-webapps/src/main/webapp/vue-app/kudos/components/KudosButton.vue]: I need to check on a string value to manage the display of the label when hovering over kudos button: "only others can send you a kudos" & "send a kudos"

File: kudos-webapps/src/main/webapp/vue-app/kudos/components/KudosButton.vue [hunk 8]
```diff
@@ -38,7 +38,7 @@
         </div>
       </template>
       <span>
-        {{ buttonDisabled && $t('exoplatform.kudos.info.onlyOtherCanSendYouKudos') || $t('exoplatform.kudos.title.sendAKudos') }}
+        {{ buttonDisabled === 'same' && $t('exoplatform.kudos.info.onlyOtherCanSendYouKudos') || buttonDisabled === 'inactive' && $t('exoplatform.kudos.title.sendAKudos') }}
```
Comment (ahamdi) [inline kudos-webapps/src/main/webapp/vue-app/kudos/components/KudosButton.vue]:
```suggestion
        {{ buttonDisabled && ((this.comment.identity.id === this.userIdentityId &&   $t('exoplatform.kudos.info.onlyOtherCanSendYouKudos')) || ((inactiveCommentOwner()  || inactiveActivityOwner()) && $t('exoplatform.kudos.title.sendAKudos')) }}
```

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: kudos-webapps/src/main/webapp/vue-app/kudos/components/KudosButton.vue (hunk 8 above)
Comment (hthabti) [reply kudos-webapps/src/main/webapp/vue-app/kudos/components/KudosButton.vue]: IMO the best solution for this case is to use the common string return for activity and comment 'inactive'/'same' to ensure proper display for the label
else if I do it so, and if the first check is true other checks will be ignored, exp "((inactiveCommentOwner  || inactiveActivityOwner) && $t('exoplatform.kudos.title.sendAKudos'))"

Comment (hthabti) [inline kudos-webapps/src/main/webapp/vue-app/kudos/components/KudosButton.vue]: {{ buttonDisabled === 'same' ? $t('exoplatform.kudos.info.onlyOtherCanSendYouKudos') : $t('exoplatform.kudos.title.sendAKudos') }}

File: kudos-webapps/src/main/webapp/WEB-INF/conf/kudos/templates/notification/mail/KudosReceiverMailPlugin.gtmpl [hunk 9]
```diff
@@ -83,7 +83,7 @@
                                           " href="$VIEW_FULL_DISCUSSION_ACTION_URL" target="_blank"><%=_ctx.appRes("Notification.label.ViewFullDiscussion")%></a>
                                       </p>
                                     <% } %>
-                                    <p style="margin: 0 0 10px; color: #999999">
+                                    <p style="margin: 0 0 10px; color: #767676">
```
Comment (ahamdi) [inline kudos-webapps/src/main/webapp/WEB-INF/conf/kudos/templates/notification/mail/KudosReceiverMailPlugin.gtmpl]: Is this change related to your PR ?

File: kudos-webapps/src/main/webapp/css/main.less [hunk 10]
```diff
@@ -186,7 +186,7 @@
 }
 
 .uiIconKudos.uiIconLightGrey {
-  color: #c6c6c6 !important;
+  color: #767676 !important;
```
Comment (ahamdi) [inline kudos-webapps/src/main/webapp/css/main.less]: The same as the comment above

File: kudos-webapps/src/main/webapp/vue-app/kudos/components/KudosButton.vue (hunk 8 above)
Comment (ahamdi) [reply kudos-webapps/src/main/webapp/vue-app/kudos/components/KudosButton.vue]: @hthabti Could you update the code with your proposed change ?

Comment (hthabti) [reply kudos-webapps/src/main/webapp/vue-app/kudos/components/KudosButton.vue]: @ahamdi proposition added

File: kudos-webapps/src/main/webapp/WEB-INF/conf/kudos/templates/notification/mail/KudosReceiverMailPlugin.gtmpl (hunk 9 above)
Comment (hthabti) [reply kudos-webapps/src/main/webapp/WEB-INF/conf/kudos/templates/notification/mail/KudosReceiverMailPlugin.gtmpl]: pr rebased

**Review by ahamdi — APPROVED** [review]
